src/utils/UrlUtils.ts

```diff
@@ -7,6 +7,15 @@
 export class UrlUtils {
     public static readParams(url: string, responseMode: "query" | "fragment" = "query"): URLSearchParams {
         if (!url) throw new TypeError("Invalid URL");
+
+        // if we're doing query, then strip off hash fragment before we parse
+        if (responseMode === "query") {
+            const hashIdx = url.indexOf("#");
+            if (hashIdx >= 0) {
+                url = url.substr(hashIdx + 1);
```

Comment:
please replace `substr` with not deprecated `substring` version

Reply:
done